hudi-common/src/main/java/org/apache/hudi/index/functional/HoodieExpressionIndex.java

```diff
@@ -59,7 +59,14 @@ public interface HoodieExpressionIndex<S, T> extends Serializable {
   String IDENTITY_FUNCTION = "identity";
 
   String EXPRESSION_OPTION = "expr";
-
+  String FORMAT_OPTION = "format";
+  String DAYS_OPTION = "days";
+  String POSITION_OPTION = "pos";
+  String LENGTH_OPTION = "len";
+  String PATTERN_OPTION = "pattern";
+  String REPLACEMENT_OPTION = "replacement";
+  String REGEX_GROUP_INDEX_OPTION = "idx";
+  String TRIM_STRING_OPTION = "trimString";
```

Comment:
Should we have this in a Map of function name to valid options? For example, `pos` option does not apply to from_unixtime but the create index command might still go through only to fail later.

Reply:
I have refactored this part. Moved these functions to separate interfaces.